File: packages/backend/src/modules/notification/notification.controller.ts
import { Controller, Get, Post, Put, Param, Query, Body, UseGuards } from '@nestjs/common';
import { NotificationService } from './notification.service';
import { Notification, NotificationType } from '../../entities/notification.entity';
import { User } from '../../entities/user.entity';
import { JwtAuthGuard } from '../../common/guards/jwt-auth.guard';
import { CurrentUser } from '../../common/decorators/current-user.decorator';

@Controller('notifications')
@UseGuards(JwtAuthGuard)
export class NotificationController {
  constructor(private notificationService: NotificationService) {}

  @Get()
  async findAll(
    @CurrentUser() user: User,
    @Query('isRead') isRead?: boolean,
    @Query('type') type?: NotificationType,
    @Query('page') page = 1,
    @Query('limit') limit = 20,
  ) {
    return this.notificationService.findByUser(
      user.id,
      { isRead: isRead !== undefined ? isRead === true : undefined, type },
      Number(page),
      Number(limit),
    );
  }

  @Get('unread-count')
  async getUnreadCount(@CurrentUser() user: User): Promise<{ count: number }> {
    const count = await this.notificationService.getUnreadCount(user.id);
    return { count };
  }

  @Put(':id/read')
  async markAsRead(@Param('id') id: string, @CurrentUser() user: User): Promise<Notification> {
    return this.notificationService.markAsRead(id, user.id);
  }

  @Post('read-all')
  async markAllAsRead(@CurrentUser() user: User): Promise<{ success: boolean }> {
    await this.notificationService.markAllAsRead(user.id);
    return { success: true };
  }
}

File: backend/src/main/java/com/gym/controller/NotificationController.java
package com.gym.controller;

import com.gym.common.ApiResponse;
import com.gym.entity.Notification;
import com.gym.service.NotificationService;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/notifications")
public class NotificationController {

    private final NotificationService notificationService;

    public NotificationController(NotificationService notificationService) {
        this.notificationService = notificationService;
    }

    @GetMapping("/user/{userId}")
    public ApiResponse<List<Notification>> getNotificationsByUser(@PathVariable Long userId) {
        return ApiResponse.success(notificationService.getNotificationsByUser(userId));
    }

    @GetMapping("/user/{userId}/unread")
    public ApiResponse<List<Notification>> getUnreadNotifications(@PathVariable Long userId) {
        return ApiResponse.success(notificationService.getUnreadNotificationsByUser(userId));
    }

    @GetMapping("/user/{userId}/count-unread")
    public ApiResponse<Long> countUnread(@PathVariable Long userId) {
        return ApiResponse.success(notificationService.countUnread(userId));
    }

    @GetMapping("/user/{userId}/page")
    public ApiResponse<Page<Notification>> getNotificationsPage(@PathVariable Long userId, Pageable pageable) {
        return ApiResponse.success(notificationService.getNotificationsByUser(userId, pageable));
    }

    @PutMapping("/{id}/read")
    public ApiResponse<Void> markAsRead(@PathVariable Long id) {
        notificationService.markAsRead(id);
        return ApiResponse.success(null);
    }

    @PutMapping("/user/{userId}/read-all")
    public ApiResponse<Void> markAllAsRead(@PathVariable Long userId) {
        notificationService.markAllAsRead(userId);
        return ApiResponse.success(null);
    }
}
